packages/graphql/src/schema/resolvers/subscriptions/where/utils/filter-relationship-key.ts

```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { Node, RelationField, RelationshipSubscriptionsEvent } from "../../../../../types";
+import type { ObjectFields } from "../../../../get-obj-field-meta";
+import type { InterfaceType, RecordType, RelationshipType, StandardType, UnionType } from "../../types";
+import { filterByProperties } from "../filters/filter-by-properties";
+import { isInterfaceSpecificFieldType, isInterfaceType, isStandardType } from "./type-checks";
+
+type EventProperties = {
+    from: Record<string, unknown>;
+    to: Record<string, unknown>;
+    relationship: Record<string, unknown>;
+};
+
+export function filterRelationshipKey({
+    receivedEventRelationship,
+    where,
+    relationshipFields,
+    receivedEvent,
+    nodes,
+}: {
+    receivedEventRelationship: RelationField;
+    where: RecordType | Record<string, RelationshipType | RecordType> | Record<string, RecordType>[];
+    relationshipFields: Map<string, ObjectFields>;
+    receivedEvent: RelationshipSubscriptionsEvent;
+    nodes: Node[];
+}): boolean {
+    const receivedEventProperties = receivedEvent.properties;
+    const receivedEventRelationshipName = receivedEventRelationship.fieldName;
+    const receivedEventRelationshipData = where[receivedEventRelationshipName] as Record<string, RelationshipType>;
+    const isRelationshipOfReceivedTypeFilteredOut = !receivedEventRelationshipData;
+    if (isRelationshipOfReceivedTypeFilteredOut) {
+        // case `actors: {}` filtering out relationships of other type
+        return false;
+    }
+    const isRelationshipOfReceivedTypeIncludedWithNoFilters = !Object.keys(receivedEventRelationshipData).length;
+    if (isRelationshipOfReceivedTypeIncludedWithNoFilters) {
+        // case `actors: {}` including all relationships of the type
+        return true;
+    }
+    const relationshipPropertiesInterfaceName = receivedEventRelationship.properties || "";
+
+    const { edge: edgeProperty, node: nodeProperty, ...unionTypes } = receivedEventRelationshipData;
+    if (
+        edgeProperty &&
+        !filterRelationshipEdgeProperty({
+            relationshipFields,
+            relationshipPropertiesInterfaceName,
+            edgeProperty,
+            receivedEventProperties,
+        })
+    ) {
+        return false;
+    }
```

Comment:
If we assign the result of the `filterRelationshipEdgeProperty()` call to a variable with a meaningful name it becomes a lot easier to read. Not sure if `hasRelationshipEdgeProperty` is correct here, but here's an example of what I mean:

```suggestion
    const hasRelationshipEdgeProperty =
        edgeProperty &&
        !filterRelationshipEdgeProperty({
            relationshipFields,
            relationshipPropertiesInterfaceName,
            edgeProperty,
            receivedEventProperties,
        });
    if (hasRelationshipEdgeProperty) {
        return false;
    }
```

Reply:
There are several instances of similar `if` cases where it isn't particularly clear what is actually being checked.